Core/include/Acts/Clusterization/DisjointSet.hpp

```diff
@@ -0,0 +1,107 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <cassert>
+#include <utility>
+#include <vector>
+
+namespace Acts::Ccl {
+using Label = std::size_t;
+constexpr Label NO_LABEL = 0;
+}  // namespace Acts::Ccl
+
+namespace Acts::Ccl {
+class DisjointSets {
+ public:
+  explicit DisjointSets(std::size_t initial_size = 128)
+      : m_defaultCapacity(initial_size), m_nextId(1) {
+    // index 0 is a dummy so that id == index
+    m_parent.reserve(m_defaultCapacity + 1);
+    m_rank.reserve(m_defaultCapacity + 1);
+    m_parent.push_back(0);
+    m_rank.push_back(0);
+  }
+  ~DisjointSets() = default;
+
+  DisjointSets(const DisjointSets&) = default;
+  DisjointSets& operator=(const DisjointSets&) = default;
+
+  DisjointSets(DisjointSets&&) noexcept = default;
+  DisjointSets& operator=(DisjointSets&&) noexcept = default;
+
+  inline Acts::Ccl::Label makeSet() {
+    // If we have reach the maximum extend the capacity if the vectors
+    // and avoid the zero-filling from a resize
+    if (m_parent.size() == m_parent.capacity()) {
+      const std::size_t newCap = m_parent.capacity() * 2;
+      m_parent.reserve(newCap);
+      m_rank.reserve(newCap);
+    }
+
+    const std::size_t id = m_nextId++;
+    m_parent.push_back(id);
+    m_rank.push_back(0);
+    return id;
+  }
+
+  inline void unionSet(std::size_t x, std::size_t y) noexcept {
+    std::size_t rootX = findRoot(x);
+    std::size_t rootY = findRoot(y);
+    if (rootX == rootY)
+      return;
```

Comment:
```suggestion
    if (rootX == rootY) {
      return;
    }
```
the linter will flag this